### ISSUE TITLE
Dynamic bones feature request

### ISSUE THREAD

**Issue by Darkspace65** [synfig/synfig#312]
Hi!
Lately I am having tons of fun experimenting with the Dynamic value type implemented by @genete. With a few complicated workarounds I managed to link shapes that are controlled by bones and a skeleton layer to shapes that are set to be dynamic. Would it be possible to add dynamic value types to the bones themselves, a bit like in Moho Pro. I think it would make things much easier. 
Greetz!

**Comment by caryoscelus** [comment]
I have only looked at your file ([ref](http://www.synfig.org/forums/viewtopic.php?f=8&t=10435)) through text editor (my current build is unable to process it, heh) and i see you're just using dynamic value node for bone angle, right? That doesn't seem to be too complicated to me, except for the fact that the whole valuenode system lacks proper UI now.

I'm not sure what you mean by adding dynamic to the bones. Simply embedding dynamic value node there even as an option would just terribly complicate both working with code and the bones themselves.

**Comment by Darkspace65** [comment]
[http://blog.smithmicro.com/4-steps-to-creating-automatic-animation-with-bone-dynamics](url) This is what I mean with adding dynamics to the bones.  In Synfig I found it complicated if not impossible to make the child bone follow the dynamics of the parent bone. That's why I ended up using the bone angle conversion and I added the force manually. Complicated is of course a matter of opinion, but I am good at drawing things. Everything that even smells like programming or exorting values to me is complicating. In Moho this takes 2 seconds. In Synfig a lot longer, crashes not included. Hence my request.

**Comment by caryoscelus** [comment]
It looks to me like what you did is not the same as what is described in that article (as far i understand, "dynamic" bones react to the whole skeleton movement), which is currently not possible in Synfig at all.

If you want something like that, it might make sense to add that to bones because it needs info of bone movement. However, it's also more complicated (probably a lot) than a stand-alone dynamics value node.

As for valuenodes currently being hard to use, i can totally agree on that, but i believe they can be made much more friendly.

**Comment by Darkspace65** [comment]
Yes. You are right: there is no info of the bone movement to the dynamic of an object right now. There is a workaround by making a dummy object that is attached to the bone with the bone link value and its center linked to the object that has dynamic values set, but like I said it's kind of complicated, lol. Maybe a revised UI might ease the pain a bit.

**Comment by caryoscelus** [comment]
Yeah, sounds complicated. I guess i should check it out to see how it actually works.

**Comment by caryoscelus** [comment]
Okay, fixed that crash and checked out your example.

So, to me the main difference between it and the moho article is amount of manual value animating (because complicated valuenode chains can be made simpler with ui and even some automation, like preset valuenode chains that can be added quickly): physics in article seem to follow motion of skeleton, while in your example dynamics are applied locally and animated manually.

To get real automatic system a lot of effort will be required to treat bones as physical objects and track their movement. It would be an interesting thing to implement, but i don't think it should be of high priority as it is more of a nice addition than a major core feature.

To make your technique more easily reproducible, better valuenode manipulation UI is needed. Since this is a more generic issue, i created a separate fr for it (https://github.com/synfig/synfig/issues/315).
